framework/base/Application.php

```diff
@@ -258,7 +258,7 @@ public function preInit(&$config)
         foreach ($this->coreComponents() as $id => $component) {
             if (!isset($config['components'][$id])) {
                 $config['components'][$id] = $component;
-            } elseif (is_array($config['components'][$id]) && !isset($config['components'][$id]['class']) && !isset($config['components'][$id]['__class'])) {
+            } elseif (is_array($config['components'][$id]) && !isset($config['components'][$id]['class'])) {
```

Comment:
And what's the deal with the `__class` key? You made [a change with this key](https://github.com/yiisoft/yii2/pull/17670/files#diff-7aab80b18ec8de0f4e9dcf3168056e40R261) not so far.

Reply:
[Refactored to ServiceLocator]( https://github.com/yiisoft/yii2/pull/17720/files?short_path=6ed2533#diff-86a8d4b20894b0affd0b4dc8828373b6R202-R206)